app/assets/javascripts/discourse/tests/unit/controllers/admin-user-badges-test.js

```diff
@@ -58,4 +61,59 @@ module("Unit | Controller | admin-user-badges", function (hooks) {
     );
     assert.deepEqual(badgeNames, sortedNames, "sorts badges by name");
   });
+
+  test("performGrantBadge", async function (assert) {
+    const GrantBadgeStub = sinon.stub(UserBadge, "grant");
+    const controller = this.owner.lookup("controller:admin-user-badges");
+    const badgeToGrant = Badge.create({
```

Comment:
btw, to create models please use the store service. for example

```suggestion
    const store = this.owner.lookup("service:store");
    // or: getOwner(this).lookup("service:store");

    const badgeToGrant = store.createRecord("badge", {
```